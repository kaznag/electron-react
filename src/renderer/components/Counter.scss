button {
  margin: 5px;
  width: 60px;
}

.count {
  display: inline-block;
  text-align: center;
  width: 100px;
}
