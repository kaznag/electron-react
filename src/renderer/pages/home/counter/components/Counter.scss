button {
  margin: 5px;
  width: 60px;
}

.count {
  display: inline-block;
  text-align: center;
  width: 100px;
}

.title {
  font-size: x-large;
  height: 40px;
}
