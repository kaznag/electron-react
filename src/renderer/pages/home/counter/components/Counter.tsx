import React from 'react';
import { useTranslation } from 'react-i18next';

import './Counter.scss';

type CounterProps = {
  count?: number;
  onIncrementClick?: (count: number) => void;
  onDecrementClick?: (count: number) => void;
  onResetClick?: () => void;
};

const defaultValue: CounterProps = {
  count: 0,
};

function Counter(props: CounterProps): JSX.Element {
  const { t } = useTranslation();
  const count = props.count ?? defaultValue.count;

  return (
    <div>
      <div className="title">{t('counter')}</div>
      <button
        onClick={() => {
          if (props.onDecrementClick) {
            props.onDecrementClick(10);
          }
        }}
        data-test-id="decrement-button-10"
      >
        -10
      </button>
      <button
        onClick={() => {
          if (props.onDecrementClick) {
            props.onDecrementClick(1);
          }
        }}
        data-test-id="decrement-button-1"
      >
        -1
      </button>
      <span className="count" data-test-id="count">
        {count}
      </span>
      <button
        onClick={() => {
          if (props.onIncrementClick) {
            props.onIncrementClick(1);
          }
        }}
        data-test-id="increment-button-1"
      >
        +1
      </button>
      <button
        onClick={() => {
          if (props.onIncrementClick) {
            props.onIncrementClick(10);
          }
        }}
        data-test-id="increment-button-10"
      >
        +10
      </button>
      <button
        onClick={() => {
          if (props.onResetClick) {
            props.onResetClick();
          }
        }}
        data-test-id="reset-button"
      >
        reset
      </button>
    </div>
  );
}

export { Counter };
