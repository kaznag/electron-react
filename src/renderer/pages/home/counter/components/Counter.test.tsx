import '@testing-library/jest-dom';

import { configure, fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import Counter from './Counter';

configure({ testIdAttribute: 'data-test-id' });

jest.mock('react-i18next', () => ({
  ...jest.requireActual('react-i18next'),
  useTranslation: () => {
    return {
      t: (str: string) => str,
    };
  },
}));

describe('<Counter />', () => {
  test('should render default Counter', () => {
    const { asFragment } = render(<Counter />);

    expect(asFragment()).toMatchSnapshot();
  });

  test('should render with count', async () => {
    await render(<Counter count={10} />);

    expect(screen.getByTestId('count')).toHaveTextContent('10');
  });

  test('should click +1 button', () => {
    const handler = jest.fn();
    render(<Counter onIncrementClick={handler} />);

    fireEvent.click(screen.getByTestId('increment-button-1'));
    expect(handler).toHaveBeenCalledTimes(1);
    expect(handler).toHaveBeenCalledWith(1);
  });

  test('should click +10 button', () => {
    const handler = jest.fn();
    render(<Counter onIncrementClick={handler} />);

    fireEvent.click(screen.getByTestId('increment-button-10'));
    expect(handler).toHaveBeenCalledTimes(1);
    expect(handler).toHaveBeenCalledWith(10);
  });

  test('should click -1 button', () => {
    const handler = jest.fn();
    render(<Counter onDecrementClick={handler} />);

    fireEvent.click(screen.getByTestId('decrement-button-1'));
    expect(handler).toHaveBeenCalledTimes(1);
    expect(handler).toHaveBeenCalledWith(1);
  });

  test('should click -10 button', () => {
    const handler = jest.fn();
    render(<Counter onDecrementClick={handler} />);

    fireEvent.click(screen.getByTestId('decrement-button-10'));
    expect(handler).toHaveBeenCalledTimes(1);
    expect(handler).toHaveBeenCalledWith(10);
  });

  test('should click reset button', () => {
    const handler = jest.fn();
    render(<Counter onResetClick={handler} />);

    fireEvent.click(screen.getByTestId('reset-button'));
    expect(handler).toHaveBeenCalledTimes(1);
  });
});
