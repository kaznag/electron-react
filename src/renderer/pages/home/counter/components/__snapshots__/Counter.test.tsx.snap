// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> should render default Counter 1`] = `
<DocumentFragment>
  <div>
    <div
      class="title"
    >
      counter
    </div>
    <button
      data-test-id="decrement-button-10"
    >
      -10
    </button>
    <button
      data-test-id="decrement-button-1"
    >
      -1
    </button>
    <span
      class="count"
      data-test-id="count"
    >
      0
    </span>
    <button
      data-test-id="increment-button-1"
    >
      +1
    </button>
    <button
      data-test-id="increment-button-10"
    >
      +10
    </button>
    <button
      data-test-id="reset-button"
    >
      reset
    </button>
  </div>
</DocumentFragment>
`;
